recipes/libs/libxml2.yaml

```diff
@@ -0,0 +1,29 @@
+inherit: [autotools]
```

Comment:
Interesting. Ralf built this package in [his PR](https://github.com/BobBuildTool/basement-gnu-linux/pull/18/commits/af14e08debc163a08c91263cc8c21a73c3f5c234) with meson. But there it lookes like it does not build out-of-tree and needs the host-toolchain. @rhubert any thought's about the right approach here?

Reply:
I can't remember why and it seams to work without as well :thinking: 
Anyway - I think meson is the more modern tool...

Reply:
I kept autotools for now, well, because it works.

Reply:
Hmm..but there is one difference: 
the meson build adds the project name to the include dir:
https://gitlab.gnome.org/GNOME/libxml2/-/blob/master/meson.build?ref_type=heads#L25

resulting in `usr/include/libxml2`. That's why I added

`CPPFLAGS+=" -I ${BOB_DEP_PATHS[libs::libxml2-dev]}/usr/include/libxml2" \` in the libxslt-recipe. cause the `cpackage` class did only pick-up `usr/include` 

https://github.com/BobBuildTool/basement-gnu-linux/pull/18/commits/3d9c8ece77a10fb3b7349ec9e7833e6bd7d00032#diff-43ce185726866de6d58028ba24f94b284b1e9514586611281393bb520c394028R25 

Switching to libxslt to `cmake` makes `CPPFLAGS` export  unnecessary.

Reply:
Hmm, as you can see I also build libxslt and it works just fine. It uses pkg-config to find the correct cflags.

Reply:
I noticed this and tried to use your libxslt in my PR but it fails with my libxml2... With the meson build the `pkg-config` does not include `the `libxml2` directory:

```
$ cat usr/lib/pkgconfig/libxml-2.0.pc 
prefix=/usr
includedir=${prefix}/include
libdir=${prefix}/lib

modules=1

Name: libXML
Description: libXML library version2.
Version: 2.13.3
Cflags: -I${includedir} -pthread
Requires:  zlib
Libs:  -L${libdir} -lxml2 -lm -ldl -pthread

$ tree usr/include/
usr/include/
└── libxml2
    └── libxml
        ├── c14n.h
[...]
```

`usr/lib/cmake/libxml2/libxml2-config.cmake` is correct:
```cmake
set(LIBXML2_INCLUDE_DIR    ${_libxml2_rootdir}/include/libxml2)
```

So building `libxml2` with `meson` and `libxslt` with `cmake` works fine. I'd prefer to use `meson` and `cmake` instead of `autotools` as I do not need to patch the recipes then when using git instead of url-scm. But there are so many patches needed in our buildenv - two more or less do not make a big difference...

Reply:
Well, I have to say there seems to be a transition phase in libxml2 where they change the build system and make changes to their configuration, however it doesn't seem to be fully finished, yet. So, again, why do you want this new stuff when it doesn't seem to work correctly and the old way of doing things, just works fine. As I said my two recipes for libxml2 and libxslt just work fine together. So what is the exact problem you are trying to solve?

Reply:
>  So what is the exact problem you are trying to solve?

In our build environment we do _NOT_ have access to the release tarballs. So what I'm trying to achieve is to have recipes where I can just add a override to switch to the git sources. Using autotools this is not possible cause there is no `configure` in the git-repo. This is only added to the release tarball. So I need more than just a `scmOverride` to make the recipe work with git-sources instead of the release tarball. Using `meson` / `cmake` I do not have this issue.

Of course your combination of version + tools also works fine and I'll just drop my libxml2+libxslt recipes if this is merged prior to my changes.

Reply:
Thanks. This explains it of course. I think this is something for Jan to decide. Normally I would always prefer in using the official release packages. On the other hand this stuff here should be easy to use and if this is a general requirement it should be at least considered.

Reply:
Correct me if I'm wrong but this is my current understanding:

* libxml2 always installs its headers in `usr/include/libxml2/libxml/...`
* Applications or other libraries do `#include <libxml/....h>
* They thus rely on that CFLAGS contains an `-I.../libxml2` option

To me it looks like the meson build produces a wrong pkgconfig file. The CMake-config file and the autotools build pkgconfig-file seems to support that. I'm in favour in switching to meson if it makes Ralfs life easier. That implies that the meson build should be fixed, preferably upstream. So I'd like to merge the autotools based package that is known to work first. I'm happy to merge a meson based recipe with a fixed pkg-config file generation anytime.

Reply:
> Using autotools the headers are installed in `usr/include/libxml/...`. That's also used by the pkg-config file.

Hmm, are you sure? I've just built libxml2 via autotools and the headers are in `usr/include/libxml2/libxml/...`. Also, `usr/lib/pkgconfig/libxml-2.0.pc` has as last line:

```
Cflags: -I${includedir}/libxml2
```

I guess this very last line is what is missing from the meson build...

Reply:
The upstream commit fixing this is:

https://gitlab.gnome.org/GNOME/libxml2/-/commit/a4517bfeaff16e5f931b258a32301d9b479ebdf6

Reply:
This is a pity. I'd really like to rely on source tarballs in the standard recipes. But patching the sources would conflict with your requirement to switch to an internal git source. Maybe we should really think about implementing https://github.com/BobBuildTool/bob/issues/347 in a way that allows to override the SCM _and_ the patches that are applied.